activesupport/lib/active_support/core_ext/string/output_safety.rb

```diff
@@ -16,7 +16,9 @@ module Util
                                 "\u{200C}-\u{200D}\u{2070}-\u{218F}\u{2C00}-\u{2FEF}\u{3001}-\u{D7FF}\u{F900}-\u{FDCF}" \
                                 "\u{FDF0}-\u{FFFD}\u{10000}-\u{EFFFF}"
     TAG_NAME_START_REGEXP = /[^#{TAG_NAME_START_REGEXP_SET}]/
-    TAG_NAME_FOLLOWING_REGEXP = /[^#{TAG_NAME_START_REGEXP_SET}\-.0-9\u{B7}\u{0300}-\u{036F}\u{203F}-\u{2040}]/
+    TAG_NAME_FOLLOWING_REGEXP_SET = "#{TAG_NAME_START_REGEXP_SET}\\-.0-9\u{B7}\u{0300}-\u{036F}\u{203F}-\u{2040}"
```

Comment:
@jhawthorn this was exactly your suggestion except I had to double-escape the `-` here (`\\-`). Using a single slash here results in an unescaped dash being passed to the regex